Sources/momento/internal/DataClient.swift

```diff
@@ -168,4 +339,426 @@ class DataClient: DataClientProtocol {
             )
         }
     }
+    
+    func listConcatenateBack(
+        cacheName: String,
+        listName: String,
+        values: [ScalarType],
+        truncateFrontToSize: Int? = nil,
+        ttl: TimeInterval? = nil
+    ) async -> CacheListConcatenateBackResponse {
+        var request = CacheClient__ListConcatenateBackRequest()
+        request.listName = Data(listName.utf8)
+        request.values = values.map(self.convertScalarTypeToData)
+        request.truncateFrontToSize = UInt32(truncateFrontToSize ?? 0)
+        request.ttlMilliseconds = UInt64(ttl ?? self.defaultTtlSeconds*1000)
+        
+        let headers = self.makeHeaders(cacheName: cacheName)
+        let call = self.client.listConcatenateBack(
+            request,
+            callOptions: CallOptions(
+                customMetadata: .init(
+                    headers.map { ($0, $1) }
+                )
+            )
+        )
+        
+        do {
+            _ = try await call.response.get()
```

Comment:
The `concatenate` and `push` operations return a `listLength` value containing the new length of the list. We need to go ahead and capture this value, supply it to the `*Success` response objects, and expose it as a `listLength` property of those objects.